unstructured_ingest/v2/processes/connectors/couchbase.py

```diff
@@ -0,0 +1,157 @@
+import json
+from dataclasses import dataclass, field
+from pathlib import Path
+from typing import TYPE_CHECKING, Any, Optional
+
+from unstructured_ingest.enhanced_dataclass import enhanced_field
+from unstructured_ingest.utils.data_prep import batch_generator
+from unstructured_ingest.utils.dep_check import requires_dependencies
+from unstructured_ingest.v2.interfaces import (
+    AccessConfig,
+    ConnectionConfig,
+    FileData,
+    UploadContent,
+    Uploader,
+    UploaderConfig,
+    UploadStager,
+    UploadStagerConfig,
+)
+from unstructured_ingest.v2.logger import logger
+from unstructured_ingest.v2.processes.connector_registry import (
+    DestinationRegistryEntry,
+)
+
+if TYPE_CHECKING:
+    from couchbase.cluster import Cluster
+
+CONNECTOR_TYPE = "couchbase"
+SERVER_API_VERSION = "1"
+
+
+@dataclass
+class CouchbaseAccessConfig(AccessConfig):
+    connection_string: str = enhanced_field(sensitive=True)
```

Comment:
Are these all required? If not, we should mark them all as optional with a default of `None` and provide some checks in a `__post_init__` method, checking that either the connection string was provided or the username/password was.

Reply:
marked as optional with a checker